Swiften/MIX/MIX.h

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2017 Tarun Gupta
+ * Licensed under the simplified BSD license.
+ * See Documentation/Licenses/BSD-simplified.txt for more information.
+ */
+
+#pragma once
+
+#include <memory>
+#include <string>
+#include <unordered_set>
+
+#include <boost/signals2.hpp>
+#include <boost/signals2/connection.hpp>
+
+#include <Swiften/Base/API.h>
+#include <Swiften/JID/JID.h>
+#include <Swiften/Elements/MIXJoin.h>
+#include <Swiften/Elements/MIXLeave.h>
+#include <Swiften/Elements/ErrorPayload.h>
+
+namespace Swift {
+    class SWIFTEN_API MIX {
+        public:
+            using ref = std::shared_ptr<MIX>;
+
+        public:
+            virtual ~MIX();
+
+            /**
+            * Join a MIX channel and subscribe to nodes.
+            */
+            virtual void joinChannelAndSubscribe(const JID &channel, const std::unordered_set<std::string>& nodes) = 0;
+            virtual void joinChannel(const JID &channel) = 0;
+
+            virtual void leaveChannel(const JID &channel) = 0;
+
+        public:
+            boost::signals2::signal<void (MIXJoin::ref, const JID&, const std::unordered_set<std::string>&)> onJoinComplete;
```

Comment:
would be nice if you document the signal parameters, like `MIXJoind::ref /* originalJoinRequest */, const JID& /* mixJID */, …`. Same goes for other signals.